### PR TITLE
Delete functions for model storage

### PR DESCRIPTION
**Pull request by mwielgus**
and some additional fields in Pod/Node/NamespaceInfo

### CODE REVIEW

**Comment by mwielgus** [comment]
cc: @vishh @piosz @jszczepkowski

File: model/impl.go [hunk 1]
```diff
@@ -57,6 +57,10 @@ func (rc *realModel) updateTime(new_time time.Time) {
 	rc.timestamp = new_time
 }
 
+func getPodKey(namespace, name string) string {
+	return namespace + "|" + name
```
Comment (piosz) [inline model/impl.go]: How about using "/" here? I believe I've seen somewhere convention <namespace>/<name>

File: model/impl.go [hunk 2]
```diff
@@ -101,6 +122,22 @@ func (rc *realModel) addNamespace(name string) *NamespaceInfo {
...
 }
 
+// Delets the given namespace with all of the pods.
+func (rc *realModel) deleteNamespace(namespace string) {
+	rc.lock.Lock()
+	defer rc.lock.Unlock()
+
+	if namespaceInfo, ok := rc.Namespaces[namespace]; ok {
+		for _, podInfo := range namespaceInfo.Pods {
+
```
Comment (piosz) [inline model/impl.go]: remove this empty line

File: model/impl.go [hunk 3]
```diff
@@ -80,6 +85,21 @@ func (rc *realModel) addNode(hostname string) *NodeInfo {
...
 }
 
+// Deletes the given node with all of the pods and containers on it.
+func (rc *realModel) deleteNode(hostname string) {
+	rc.lock.Lock()
+	defer rc.lock.Unlock()
+
+	if node, ok := rc.Nodes[hostname]; ok {
+		for _, podInfo := range node.Pods {
+			if namespaceInfo, ok := rc.Namespaces[podInfo.Namespace]; ok {
```
Comment (piosz) [inline model/impl.go]: How about reusing deletePod here?

**Comment by piosz** [comment]
LGTM, just a few nits

File: model/impl.go (hunk 1 above)
Comment (mwielgus) [inline model/impl.go]: Done

File: model/impl.go (hunk 2 above)
Comment (mwielgus) [inline model/impl.go]: Done

File: model/impl.go (hunk 3 above)
Comment (mwielgus) [inline model/impl.go]: Both deleteNode and pod operate on the same lock. Here the lock is already taken and deletePod would deadlock (at least i think so - mutex doc doesn't say anything about possibility to acquire a lock multiple times - "Lock locks m. If the lock is already in use, the calling goroutine blocks until the mutex is available.")

Comment (piosz) [inline model/impl.go]: That's fine. The problem may be solved by delegating the responsibilty of locking to the caller, but maybe this is won't simplify code here.

**Comment by piosz** [comment]
LGTM

**Comment by mwielgus** [comment]
One of the tests is flaky - see: #523

**Comment by vishh** [comment]
cc @afein

**Comment by afein** [comment]
LGTM

**Pull request merged**